otx/algorithms/segmentation/adapters/mmseg/configurer.py

```diff
@@ -0,0 +1,618 @@
+"""Base configurer for mmseg config."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import importlib
+import json
+import os
+from typing import Any, Dict
+
+import numpy as np
+import torch
+from mmcv.runner import CheckpointLoader
+from mmcv.utils import Config, ConfigDict
+from torch import distributed as dist
+
+from otx.algorithms.common.adapters.mmcv.utils import (
+    align_data_config_with_recipe,
+    build_dataloader,
+    build_dataset,
+    patch_adaptive_interval_training,
+    patch_default_config,
+    patch_early_stopping,
+    patch_fp16,
+    patch_persistent_workers,
+    patch_runner,
+)
+from otx.algorithms.common.adapters.mmcv.utils.config_utils import (
+    recursively_update_cfg,
+    remove_custom_hook,
+    update_or_add_custom_hook,
+)
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.segmentation.adapters.mmseg.utils import (
+    patch_datasets,
+    patch_evaluation,
+)
+
+logger = get_logger()
+
+
+# pylint: disable=too-many-public-methods
+class SegmentationConfigurer:
+    """Patch config to support otx train."""
+
+    def __init__(self):
+        self.task_adapt_type = None
+        self.task_adapt_op = "REPLACE"
+        self.org_model_classes = []
+        self.model_classes = []
+        self.data_classes = []
+
+    # pylint: disable=too-many-arguments
+    def configure(
+        self,
+        cfg,
+        model_ckpt,
+        data_cfg,
+        training=True,
+        subset="train",
+        ir_options=None,
+        data_classes=None,
+        model_classes=None,
+    ):
+        """Create MMCV-consumable config from given inputs."""
+        logger.info(f"configure!: training={training}")
+
+        self.configure_base(cfg, data_cfg, data_classes, model_classes)
+        self.configure_device(cfg, training)
+        self.configure_model(cfg, ir_options)
+        self.configure_ckpt(cfg, model_ckpt)
```

Comment:
A recent fix requires this function to be run before configure_model.

Reply:
Updated, thank you!